### ISSUE TITLE
Help with false positives

### ISSUE THREAD

**Issue by Voicech** [AlexeyAB/darknet#2212]
Hello,

I have trained yolo v3 tiny with two classes: for detecting bicycles and pedestrians. Training size was around 4000 pictures for each class, trained for around 20000 iterations.

In certain cases yolo fails giving me a false positive (see picture attached). Surprisingly, it fails on the background image, and the prediction of that fail positive is 90%. To mention - I was training using the pictures from the same camera, with the same background, just annotating bicycles/pedestrians on it. 

How to deal with such issues - create a third class and name it "fails"? Also, I have issues with snow cleaning machines and kid trolleys, which sometimes are detected as bicycles. What's the best approach to avoid such cases? Creating a separate class for each type of a fail? I considered one approach mentioned somewhere in the forum, by adding false positives to the training set with empty annotation files. I tried it but cannot say it helped.

Any help much appreciated.

![im1](https://user-images.githubusercontent.com/46717525/51188642-07453980-18e7-11e9-985d-083b46e5ab63.jpg)

**Comment by AlexeyAB** [comment]
> Repeating my question - what to do with trolleys, snow machines, etc.? Feed as negative samples or create a separate class for them?

Feed as negative samples. As described here: https://github.com/AlexeyAB/darknet#how-to-improve-object-detection

Your training dataset should be similar to the test (production) images as much as possible.

Also check all your images by using Yolo_mark.

"Camera 01", "Person 1", "06-05-1970 ..."  - it can affect on false negatives, if you doesn't provide sufficient negative samples with similar  "Camera 01", "Person 1", "06-05-1970 ..."

**Comment by Voicech** [comment]
Hi again,

I a preparing for my new training, and the question is: you could see my production environment in the pictures above. 

Previously I was training on the samples collected from that camera, and marking a label strictly around the bicycle. However, as you could see in the example below, I must label also some parts of the rider (especially legs). As a result, I also get false positives while pedestrians passing by, as Yolo sometimes sees bicycle in their legs.
![vlcsnap-error525](https://user-images.githubusercontent.com/46717525/51733144-1798a980-2089-11e9-84db-568da259addf.png)

Is it better to use super clean pictures for training like in the picture below? What about the background of such picture? Is it important to train with the real background? Help much appreciated.
![1e2](https://user-images.githubusercontent.com/46717525/51733244-69d9ca80-2089-11e9-8ea6-a9ff130c85d9.jpg)

**Comment by AlexeyAB** [comment]
Hi,

> Is it better to use super clean pictures for training like in the picture below? 

Better to train with original images from a camera (not super clean).
To avoid false-positives, just include images with false-positives to  your Training dataset (without labels on false-positive objects).

> What about the background of such picture? 
> Is it important to train with the real background? 

Yes.

**Comment by Voicech** [comment]
Thanks so much!

**Comment by litingsjj** [comment]
@AlexeyAB Sorry to bother you! I have questions about false-positive. When I found some false-positive in some images with finished model. I put  images with false-positive(without labels on false-positive objects) to training dataset. Here I have two questions:
1. About images number with false-poisitive, if I detect only person, now I detect bicycle as person, Am I put images include bicycle as the same number as train dataset? If I detect others as person again, what should I do? Put images include other thing again? But now the the images with false-positive is larger than training dataset.  If I reduce number of images include bicycle, is it influence false detect bicycle?

2. Am I need to retrain network or finetune origin model? If finetune, how to set  parameters about lr, iteration or something else? 
I'm appreciate if you have time to give me some advice!
